pkg/controller/csr_check.go

```diff
@@ -195,7 +195,8 @@ func authorizeCSR(
 	if !ok {
 		klog.Errorf("%v: Serving Cert: No target machine for node %q", req.Name, nodeAsking)
 		//TODO: set annotation/emit event here.
-		return false, nil
+		// Return error so we requeue in case we're racing with node linker.
+		return false, fmt.Errorf("Unable to find machine for node")
```

Comment:
In a scenario where the Node has come from a Kubelet that isn't managed by Machine API, (think UPI, or BYO bare metal into a vSphere, etc), this will cause the CSR approver to start hot looping trying to approve this CSR which will never be approvable.

I think it is appropriate to requeue yes, but ideally that would be requeue after a few minutes, so we try again within a reasonable time frame, but don't hot loop

Reply:
Should not exponential back-off on requeues manage such scenario?

Reply:
There is exponential backoff yes, but that not only affects this individual resource, but all resources. The default rate limiter used in controller runtime has an overall limit and an individual item limit. So if there were a mixture of Nodes, some backed by Machines, some not, the backoff incurred by the no-Machine Nodes, could have other affects as well.

If we are confident that this won't be an issue, then that's ok, we can proceed as is, but someone should check this